src/components/views/donate/CryptoDonation.tsx

```diff
@@ -266,19 +270,18 @@ const CryptoDonation: FC = () => {
 
 	const pollToken = useCallback(async () => {
 		clearPoll();
-		if (walletChainType === ChainType.SOLANA) {
-			return setSelectedTokenBalance(
-				BigInt(Number(balance || 0) * LAMPORTS_PER_SOL),
-			);
-		}
 
 		if (!selectedToken) {
 			return setSelectedTokenBalance(0n);
 		}
 		// Native token balance is provided by the Web3Provider
 		const _selectedTokenSymbol = selectedToken.symbol.toUpperCase();
 		const nativeCurrency =
-			config.EVM_NETWORKS_CONFIG[networkId!]?.nativeCurrency;
+			config.NETWORKS_CONFIG[
+				!walletChainType || walletChainType == ChainType.EVM
+					? networkId
+					: walletChainType
+			]?.nativeCurrency;
```

Comment:
We can detect it's a solana native token by comparing the token address with the system program id, as mentioned in another comment. If that matched, useGeneralWallet has computed the balance.

Reply:
Here I've used the balance fetched from "useGeneralWallet" as you said (L:287).
It's better to avoid detecting Solana here. We would better avoid adding "if"s for each chain. I think it's better to detect chains automatically without adding conditions. e.g., here, I've removed below Solana condition, and it's working as expected:
```
if (address !== SystemProgram.programId.toBase58()) {
			throw new Error(
				'Token address is not native token address - not supported yet!',
			);
		}
```